src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -267,16 +272,25 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
             {                
                 return await asyncConnectorTexlFunction.InvokeAsync(args, _services, _cancellationToken).ConfigureAwait(false);
             }
-            else if (func is UserDefinedTexlFunction udtf)
-            {
-                // $$$ Should add _runtimeConfig
-                result = await udtf.InvokeAsync(args, _cancellationToken, context.StackDepthCounter.Increment()).ConfigureAwait(false);
-            }
             else if (func is CustomTexlFunction customTexlFunc)
             {
                 // If custom function throws an exception, don't catch it - let it propagate up to the host.
                 result = await customTexlFunc.InvokeAsync(FunctionServices, args, _cancellationToken).ConfigureAwait(false);
             }
+            else if (func is UserDefinedFunction userDefinedFunc)
+            {
+                userDefinedFunc.BindBody(_symbolValues.SymbolTable, new Glue2DocumentBinderGlue(), bindingConfig: BindingConfig.Default);

```

Comment:
> userDefinedFunc.BindBody( [](http://example.com/codeflow?start=15&length=26)

we need to move this call up. All binding should be done before we begin eval.

Reply:
I changed the approach here. Looking forward for your opinion.